### DISCUSSION TITLE
mutation not updating cache?

### DISCUSSION THREAD

**Discussion by chrisdrackett** [urql-graphql/urql#1057] [Q&A] [answered]
We have a simple mutation we're using just to wrap out heads around urql. We're using `offlineExchange`.

```
const addTaskMutation = `
  mutation ($id: ID!, $title: String!) {
    addTask (id: $id, title: $title) {
      __typename
      id
      title
    }
  }
`
```

when we run this and it successfully comes back we don't see the related task in the cache. It seems the only thing we see in the cache is the result of any queries we've run. We expected as described [here](https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/). Any thoughts on what we might be missing?

**Comment by JoviDeCroock** [comment, marked as answer]
Adding and removing entities from the cache needs more context, we can't assume where this entity belongs, this is written out in more detail [here](https://formidable.com/open-source/urql/docs/graphcache/custom-updates/)

**Reply by chrisdrackett** [reply]
I was under the assumption that given a `__typename` and `id` that the data should be added to the cache, no?

**Reply by JoviDeCroock** [reply]
I mean we can't assume what root type this entity belongs to, we can't assume that this is a direct child of an entity or from query. If it is from query we are not aware we need certain access rights/more parameters so making this assumption would make us stray away from the source of truth (your server).

So basically we'd be adding an entity to the cache that is disconnected from all links, since we don't know what links to this entity. This makes our garbage collection clean this entity.

**Reply by special-character** [reply]
I am still a bit fuzzy on why the response from the mutation isn't keyed in the cache when it has a typename and id. i.e. when I have a mutation for `addTodo`, the response doesn't key `Todo:1` in the normalized cache. Is that because it isn't linked to anything and gets garbage collected? 

I was expecting it to looks something like the docs have for the normalized cache:
```
{
...
  "Todo:1": {
    "__typename": "Todo",
    "id": "1", // record
    "title": "implement graphcache", // record
    "author": "Author:1" // link
  },
...
}
```

**Reply by JoviDeCroock** [reply]
Yes, nothing is connecting to your `Todo:1` (normally would be an existing entity that eventually leads up to a rootField like Query)

**Reply by chrisdrackett** [reply]
ok, based on this I'll probably open a new issue specifying what we want to do, but seems like maybe we need to adjust our mental model (and maybe urql does not fit).

**Reply by JoviDeCroock** [reply]
@chrisdrackett if you describe your issue we could most likely help but most of it is also described in the docs I linked about custom-updates.
